### ISSUE TITLE
How to make a selector for a directive with only an Output event

### ISSUE THREAD

**Issue by kasperpeulen** [angular/angular#6394]
I have the following directive.

```
@Directive(selector: '[longpress]')
class LongPress {
  @Output() EventEmitter<KeyboardEvent> longpress =
      new EventEmitter<KeyboardEvent>();

  KeyboardEvent _startEvent;
  @Input() Duration longpressDelay = new Duration(milliseconds: 600);
  Timer timer;

  @HostListener("keydown", const [r"$event"])
  @HostListener("keyup", const [r"$event"])
  void keyEventHandler(KeyboardEvent event) {
    if (_startEvent == null ||
        _startEvent.type != event.type ||
        _startEvent.keyCode != event.keyCode) {
      if (timer != null) {
        timer.cancel();
      }
      _startEvent = event;
      timer = new Timer(longpressDelay, () {
        longpress.add(_startEvent);
        timer = null;
      });
    }
  }
}
```

I want it to use like:

```
<textarea (longpress)="onLongPress($event)">
```

But it seems angular forces me to use it like this:

```
<textarea longpress (longpress)="onLongPress($event)">
```

Which seems a bit awkward to me. Is there any selector for events?

**Comment by johnrbaur** [comment]
I suspect you could change your first line to 

```@Directive(selector: '[(longpress)]')```

If you add a plunkr, I could help you test it.

**Comment by zoechi** [comment]
There is nothing like Plunker for Dart (at least nothing that can be used with Angular2).  
I tried `@Directive(selector: '[(longpress)]')` but the directive is not instantiated with this selector.

**Comment by ericmartinezr** [comment]
You can't. This [pull request](https://github.com/angular/angular/pull/5914/commits) won't let you.

**Comment by ericmartinezr** [comment]
@kasperpeulen this https://github.com/angular/angular/pull/6870 allows you to do this now

**Comment by DrewynV** [comment]
__________________________________

Hi! We just wanted to let you know we receievd your email. Thank you for submitting your request. Our aim is to get back to you within 24 hours at the latest. If you have any additional information to add to your request, please reply to this email.


Also please visit our knowledgebase (http://www.support.magicdust.com.au) to browse for further info, we are adding to it all the time.

Or here is our industry report about the 11 Essentials for a Successful Website (http://www.magicdust.com.au/web-design/.

We look forward to helping you succeed online.

Thanks

The Team @ Magicdust Digital

http://support.magicdust.com.au/

Ph (02) 9979 7474
M-F | 9 am-5:00 pm AEST (AEDT in Summer) excluding Public Holidays

------------------------------------------------------------------------------------------------------------
This message was sent to  reply@reply.github.com in reference to Case #166494.
------------------------------------------------------------------------------------------------------------


[[698f895a98b0d89bd99da864ad4f29780f8840ce-673073812]]

**Comment by pkozlowski-opensource** [comment]
Already fixed in  231773e
